### PR TITLE
NF: refactorings for critical exceptions

### PR DESCRIPTION
**Pull request by david-allison-1**
Reviewer: These touch analytics, please review harshly

## Purpose / Description
Pruning off refactoring fixes from #9184

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison-1** [comment]
blocked on #9223

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/Acra.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ */
+package com.ichi2.anki.analytics
+
+import android.content.Context
+import com.ichi2.anki.AnkiDroidApp
+
+class Acra {
+    companion object {
+
+        fun enableWithAskDialog(ctx: Context) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/analytics/Acra.kt:24]: Am I missing something about how Kotlin works? I must admit I have not read up on it too much and am not sure about companion objects, but this method and the `disableAcra` appear unused at this point?

**Review by mikehardy — APPROVED** [review]
Nothing harsh to say about it, but I had a question about possibly unused (at this point) methods that may just be me not knowing kotlin yet. Not a blocker either way

**Comment by mikehardy** [comment]
If it were rebased over current master the unit test would pass (it's the lint check for kotlin companion objects) and the emulator test was the new/old friend deck picker click counts. In other words, CI status is not a blocker for merge at the moment

**Comment by mikehardy** [comment]
This is marked as draft but is merge-worthy as is, IMHO, please feel free

File: AnkiDroid/src/main/java/com/ichi2/anki/analytics/Acra.kt (hunk 1 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/analytics/Acra.kt:24]: They're unused, adding capabilities to be used later.

**Review by Arthur-Milchior — APPROVED** [review]
I don't know what is the ultimate purpose, but that seems quite NF for me to approve.

I'll let you merge yourself

**Comment by mikehardy** [comment]
@Arthur-Milchior this is building towards the ability to take certain really important exceptions that are not crashes (so we don't need ACRA) and send them to analytics every time instead of filtered through our reduced-sending sampling strategy filter (which is set to 10% of users, to avoid overwhelming google analytics free tier) and through our de-duplication filter (where the same stack is never re-sent)

The purpose is to take things where we aren't sure if they are frequent or not and get real data on their frequency for people with usage stats enabled, I believe the Rust library loading (or not!) is something we want to know *exactly* how frequent it is, and if it is self-resolving or not

@david-allison-1 glad I wasn't misunderstanding the kotlin completely :-) - future APIs, works for me

**Pull request merged**